### DISCUSSION TITLE
Validation loss calculation with MONAI engines

### DISCUSSION THREAD

**Discussion by KSchmidtACR** [Project-MONAI/tutorials#398] [Q&A]
Hello,
I have a training script which uses the the `SupervisedTrainer` and `SupervisedEvaluator`, but I would like to be able to calculate validation loss. Looking at the `SupervisedEvaluator` script, loss is not calculated. Is there a way to calculate validation loss using a MONAI engine?

**Comment by Nic-Ma** [comment]
Hi @KSchmidt2ACR ,

Thanks for your interest here.
Actually, you can set loss function as the metrics for evaluator, and define the function to compare metrics:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/engines/workflow.py#L66
If you also want to compare metrics then save checkpoint, you need to set `key_metric_negative_sign`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/checkpoint_saver.py#L66

Thanks.
